### ISSUE TITLE
rustdoc replaces `##` with `#` even if it is in a code block

### ISSUE THREAD

**Issue by mamekoro** [rust-lang/rust#118027]
I tried this code:

```rust
/// ```
/// define_component!(
///     #[derive(Debug)]
///     ##[min(0.0))]
///     ##[max(100.0))]
///     ##[default(50.0)]
///     pub Fuel(pub f32);
/// );
/// ```
macro_rules! define_component {
    (
        $(#[$attr:meta])*
        ##[min($min:expr)]
        ##[max($max:expr)]
        ##[default($default:expr)]
        $self_vis:vis $self:ident($inner_vis:vis $inner:ty);
    ) => {};
}
```

The `define_component` macro uses a single hash sign `#` to represent a normal attribute (e.g. `#[derive(Debug)]`), and uses a double hash sign `##` to represent an argument specific to that macro (e.g. `##[min(0.0)]`).

I ran `cargo doc --document-private-items --open` and opened the generated `define_component` page.

I expected to see this happen:

```text
define_component!(
    #[derive(Debug)]
    ##[min(0.0))]
    ##[max(100.0))]
    ##[default(50.0)]
    pub Fuel(pub f32);
);
```

Instead, this happened:

```text
define_component!(
    #[derive(Debug)]
    #[min(0.0))]
    #[max(100.0))]
    #[default(50.0)]
    pub Fuel(pub f32);
);
```

`##` were replaced by `#`. (e.g. `##[min(0.0)]` became `#[min(0.0)]`)

I think rustdoc should keep code blocks as written (except for coloring, etc.). This is because code blocks are generally expected to be enclosed in the HTML `<pre>` tag, which defines preformatted text.

In other words, code blocks are preformatted and should be kept as written.

Perhaps the following code is related:
https://github.com/rust-lang/rust/blob/2831701757eb7b3105eda26a306c2f3a97e2664b/src/librustdoc/html/markdown.rs#L170-L172

### Workaround
If the doc comment starts with ` ```text ` instead of ` ``` `, this problem does not occur.

### Meta
`rustdoc --version --verbose`:
```
rustdoc 1.74.0 (79e9716c9 2023-11-13)
binary: rustdoc
commit-hash: 79e9716c980570bfd1f666e3b16ac583f0168962
commit-date: 2023-11-13
host: x86_64-unknown-linux-gnu
release: 1.74.0
LLVM version: 17.0.4
```

**Comment by GuillaumeGomez** [comment]
It's because you can hide lines of code if you prepend them with `#`. There is some "intelligence" by using the parser to know when the `#` is actually used in the code or not. In the current example, the parser doesn't know and rustdoc simply removes it. I don't really see a way to fix it. An idea could be to introduce a new codeblock tag which would allow to say "keep the code as is".
